internal/webservice/routes.go

```diff
@@ -239,11 +241,55 @@ func (ws *WebService) getQueuesPerPartition(req *restful.Request, resp *restful.
 		errorResponse(req, resp, err)
 		return
 	}
-	daoQueues := make([]*dao.PartitionQueueDAOInfo, 0, len(queues))
+	root, err := buildPartitionQueueTrees(ctx, queues)
+	if err != nil {
+		errorResponse(req, resp, err)
+	}
+	jsonResponse(resp, root)
+}
+
+func buildPartitionQueueTrees(ctx context.Context, queues []*model.PartitionQueueDAOInfo) ([]*dao.PartitionQueueDAOInfo, error) {
```

Comment:
Just a quick question: Wouldn't it be better to return `[]*model.PartitionQueueDAOInfo` since it includes additional fields like created_at and deleted_at, which could be useful for the frontend when implementing designs to highlight active or deleted queues?

Reply:
Right, then we would take the hybrid approach, I'll work on that, but it does make sense to me.

Reply:
Yeah, today or tomorrow we have to support additional fields for our use case.